go/private/rules/binary.bzl

```diff
@@ -521,32 +525,35 @@ exit /b %GO_EXIT_CODE%
                 transitive = [sdk.headers, sdk.srcs, sdk.tools],
             ),
             toolchain = None,
-            outputs = [out, gotmp],
+            outputs = [out, ctx.outputs.out_pack, gotmp],
             mnemonic = "GoToolchainBinaryBuild",
         )
     else:
-        # -a flag instructs the compiler to not read from GOCACHE and force a rebuild.
-        # This provides extra safety in cases of unsandboxed execution.
+        # We do not use -a here as the cache drastically reduces the time spent
+        # on the second go build.
         args = ctx.actions.args()
-        args.add("build")
-        args.add("-a")
-        args.add("-o", out)
-        args.add("-trimpath")
-        args.add("-ldflags", ctx.attr.ldflags, format = '-buildid="" %s')
+        args.add(ctx.outputs.out_pack)
+        args.add(out)
```

Comment:
This part is confusing to me.

Essentially we have `args = out_pack + out + srcs` and 2 `go build` commands:

- `go build -o "$1" cmd/pack` which essentially mean `-o out_pack cmd/pack` which makes sense

- `go build -o "$@"` which I think meant to do `-o out <srcs>` but will be turned into `-o out_pack out <srcs>` instead?

Could we just place `out_pack` and `out` through `format()` and `srcs` through `$@` here?